test/functional/apps/dashboard/_dashboard_queries.js

```diff
@@ -0,0 +1,97 @@
+      // This flip between apps fixes the url so state is preserved when switching apps in test mode.
+      // Without this flip the url in test mode looks something like
+      // "http://localhost:5620/app/kibana?_t=1486069030837#/dashboard?_g=...."
+      // after the initial flip, the url will look like this: "http://localhost:5620/app/kibana#/dashboard?_g=...."
+      await PageObjects.header.clickVisualize();
+      await PageObjects.header.clickDashboard();
+
+      await PageObjects.dashboard.clickNewDashboard();
+      await PageObjects.dashboard.setTimepickerInDataRange();
+
+      await PageObjects.dashboard.addVisualizations([PIE_CHART_VIS_NAME]);
+      await PageObjects.dashboard.clickEditVisualization();
+      await PageObjects.dashboard.setQuery('memory:<80000');
+      await PageObjects.dashboard.clickFilterButton();
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      await PageObjects.visualize.saveVisualization(PIE_CHART_VIS_NAME);
+
+      await PageObjects.header.clickDashboard();
+
+      const pieSlicesCount = await PageObjects.dashboard.getPieSliceCount();
+      expect(pieSlicesCount).to.equal(2);
+    });
+
+    it('Pie chart attached to saved search filters data as expected', async () => {
+      await dashboardVisualizations.createAndAddSavedSearch({
+        name: 'bytes < 90',
+        query: 'bytes:<90',
+        fields: ['bytes']
+      });
+
+      await PageObjects.header.clickDashboard();
+      await PageObjects.dashboard.gotoDashboardLandingPage();
+      await PageObjects.dashboard.clickNewDashboard();
+
+      await PageObjects.dashboard.clickAddVisualization();
+      await PageObjects.dashboard.clickAddNewVisualizationLink();
+      await PageObjects.visualize.clickPieChart();
+      await PageObjects.visualize.selectSearch('bytes < 90');
+      await PageObjects.visualize.clickBucket('Split Slices');
+      await PageObjects.visualize.selectAggregation('Terms');
+      await PageObjects.visualize.selectField('memory');
+      await PageObjects.visualize.clickGo();
+      await PageObjects.visualize.saveVisualization('memory with bytes < 90 pie');
+      await PageObjects.header.clickToastOK();
+
+      let pieSlicesCount = await PageObjects.dashboard.getPieSliceCount();
+      expect(pieSlicesCount).to.equal(3);
+
+      await PageObjects.dashboard.setQuery('bytes:>100');
+      await PageObjects.dashboard.clickFilterButton();
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      // There appears to be a slight delay between the page loading and the visualization reacting to the
+      // filter which is causing this to sometimes be greater than 0, hence the retry.
+      retry.try(async () => {
+        pieSlicesCount = await PageObjects.dashboard.getPieSliceCount();
+        expect(pieSlicesCount).to.equal(0);
+      });
+    });
+
+    it('Pie chart attached to saved search filters shows no data with conflicting dashboard query', async () => {
```

Comment:
How is this test different than the one above it - specifically lines 77-86?

Reply:
Whoa, good catch, will clean this up